Sources/GRPC/LengthPrefixedMessageWriter.swift

```diff
@@ -44,25 +44,27 @@ internal struct LengthPrefixedMessageWriter {
   /// Writes the data into a `ByteBuffer` as a gRPC length-prefixed message.
   ///
   /// - Parameters:
-  ///   - message: The serialized Protobuf message to write.
+  ///   - payload: The payload to serialize and write.
   ///   - buffer: The buffer to write the message into.
   /// - Returns: A `ByteBuffer` containing a gRPC length-prefixed message.
   /// - Precondition: `compression.supported` is `true`.
   /// - Note: See `LengthPrefixedMessageReader` for more details on the format.
-  func write(_ message: Data, into buffer: inout ByteBuffer, disableCompression: Bool = false) throws {
-    buffer.reserveCapacity(LengthPrefixedMessageWriter.metadataLength + message.count)
-
+  func write(_ payload: GRPCPayload, into buffer: inout ByteBuffer, disableCompression: Bool = false) throws {
+    buffer.reserveCapacity(MemoryLayout.size(ofValue: payload) + LengthPrefixedMessageWriter.metadataLength)
```

Comment:
`MemoryLayout.size(ofValue: payload)` doesn't actually help us here since it isn't the serialised size of the payload. Just the metadata length is absolutely fine here.

Reply:
@glbrntt I actually used this because there was a test case crash where ReserveCapacity won’t really reserve enough bytes and the bytebuffer would have a precondition crash while trying to move the writer. Hence, the earlier code where I would write the bytes and then go back and replace them